sed/loader/flash/loader.py

```diff
@@ -41,11 +44,129 @@ class FlashLoader(BaseLoader):
     def __init__(self, config: dict) -> None:
 
         super().__init__(config=config)
+        self.multi_index = ["trainId", "pulseId", "electronId"]
         self.index_per_electron: MultiIndex = None
         self.index_per_pulse: MultiIndex = None
-        self.parquet_names: List[Path] = []
         self.failed_files_error: List[str] = []
 
+    def initialize_paths(self) -> Tuple[List[Path], Path]:
+        """
+        Initializes the paths based on the configuration.
+
+        Returns:
+            Tuple[List[Path], Path]: A tuple containing a list of raw data directories
+            paths and the parquet data directory path.
+
+        Raises:
+            ValueError: If required values are missing from the configuration.
+            FileNotFoundError: If the raw data directories are not found.
+        """
+        # Parses to locate the raw beamtime directory from config file
+        if "paths" in self._config["core"]:
+            data_raw_dir = [
+                Path(self._config["core"]["paths"].get("data_raw_dir", "")),
+            ]
+            data_parquet_dir = Path(
+                self._config["core"]["paths"].get("data_parquet_dir", ""),
+            )
+
+        else:
+            try:
+                beamtime_id = self._config["core"]["beamtime_id"]
+                year = self._config["core"]["year"]
+                daq = self._config["dataframe"]["daq"]
+            except KeyError as exc:
+                raise ValueError(
+                    "The beamtime_id, year and daq are required.",
+                ) from exc
+
+            beamtime_dir = Path(
+                self._config["dataframe"]["beamtime_dir"][self._config["core"]["beamline"]],
+            )
+            beamtime_dir = beamtime_dir.joinpath(f"{year}/data/{beamtime_id}/")
+
+            # Use pathlib walk to reach the raw data directory
+            data_raw_dir = []
+            raw_path = beamtime_dir.joinpath("raw")
+
+            for path in raw_path.glob("**/*"):
+                if path.is_dir():
+                    dir_name = path.name
+                    if dir_name.startswith("express-") or dir_name.startswith(
+                        "online-",
+                    ):
+                        data_raw_dir.append(path.joinpath(daq))
+                    elif dir_name == daq.upper():
+                        data_raw_dir.append(path)
+
+            if not data_raw_dir:
+                raise FileNotFoundError("Raw data directories not found.")
+
+            parquet_path = "processed/parquet"
+            data_parquet_dir = beamtime_dir.joinpath(parquet_path)
```

Comment:
This whole branch is not being tested for. I think it would be good to write a separate test for it, but could be done later.

Reply:
Yes, soon we have some data publicly accessible so I can test this with much more ease than would be possible elsewise.

Reply:
refereced in #154